src/Simulations/VarianceDissipationComputations/VarianceDissipationComputations.jl

```diff
@@ -0,0 +1,165 @@
+    Vⁿ⁻¹ = c_grid_vector(grid) 
+    Fⁿ   = c_grid_vector(grid) 
+    Fⁿ⁻¹ = c_grid_vector(grid) 
+    cⁿ⁻¹ = CenterField(grid)
+
+    previous_state   = (; cⁿ⁻¹, Uⁿ⁻¹, Uⁿ)
+    advective_fluxes = (; Fⁿ, Fⁿ⁻¹)
+    diffusive_fluxes = (; Vⁿ, Vⁿ⁻¹)
+
+    gradients = deepcopy(P)
+
+    return VarianceDissipation(P, K, advective_fluxes, diffusive_fluxes, previous_state, gradients, tracer_name)
+end
+
+function (ϵ::VarianceDissipation)(model)
+
+    # Check if the model is using a QuasiAdamsBashforth2 time stepper
+    if !(model.timestepper isa QuasiAdamsBashforth2TimeStepper)
+        throw(ArgumentError("VarianceDissipation is only supported for QuasiAdamsBashforth2 time-stepping."))
+    end
+
+    # Check if the model has a velocity field
+    if !hasproperty(model, :velocities)
+        throw(ArgumentError("Model must have a velocity field."))
+    end
+
+    # Check if the model has tracers
+    if !hasproperty(model.tracers, ϵ.tracer_name)
+        throw(ArgumentError("Model must have a tracer called $tracer_name."))
+    end
+
+    # First we compute the dissipation from previously computed fluxes
+    compute_dissipation!(model, ϵ, ϵ.tracer_name)
+
+    # Then we update the fluxes to be used in the next time step
+    cache_fluxes!(model, ϵ, ϵ.tracer_name)
+    
+    return nothing
+end
+
+@inline getadvection(advection, tracer_name) = advection
+@inline getadvection(advection::NamedTuple, tracer_name) = @inbounds advection[tracer_name]
+
+const f = Face()
+const c = Center()
+
+include("update_fluxes.jl")
+include("advective_dissipation.jl")
+include("diffusive_dissipation.jl")
+include("compute_dissipation.jl")
+include("flatten_dissipation_fields.jl")
+
+# A VarianceDissipation object requires a `ConsecutiveIteration` schedule to make sure
+# that the computed fluxes are correctly used in the next time step.
+# Also, the `VarianceDissipation` object needs to be called on `UpdateStateStepCallsite` to be correct.
+function Callback(func::VarianceDissipation, schedule;
+                  parameters = nothing,
+                  callsite = UpdateStateCallsite())
+
+    if !(callsite isa UpdateStateCallsite)
+        @warn "VarianceDissipation callback must be called on UpdateStateCallsite. Changing `callsite` to `UpdateStateCallsite()`."
+        callsite = UpdateStateCallsite()
+    end
+
+    if schedule isa TimeInterval # Time step might change size between iterations invalidating `VarianceDissipation`
+        throw(ArgumentError("a VarianceDissipation computation must be executed on `IterationInterval`s. \n" *
+                            "The provided `TimeInterval` schedule is not supported"))
+    end
+
+    if !(schedule isa ConsecutiveIterations || schedule == IterationInterval(1))
+        @warn "VarianceDissipation callback must be called every Iteration or on `ConsecutiveIterations`. \n" * 
+              "Changing `schedule` to `ConsecutiveIterations(schedule)`."
+        schedule = ConsecutiveIterations(schedule)
+    end
```

Comment:
I suggest using dispatch to handle this

Reply:
I can add a validation function that I can dispatch on.